lib/fog/azurerm/models/network/network_security_group.rb

```diff
@@ -48,6 +48,32 @@ def destroy
           service.delete_network_security_group(resource_group, name)
         end
 
+        def update(hash = {})
+          raise('Provided hash can not be empty.') if hash.empty? || hash.nil?
+
+          if !hash[:name].nil? || !hash[:location].nil? || !hash[:resource_group].nil? || !hash[:id].nil?
```

Comment:
I've replaced this method name with `update_security_rule`. So now we don't need to put these checks in it.